packages/delisp-core/src/unify.ts

```diff
@@ -33,7 +34,15 @@ interface UnifyMismatchError {
   t2: Monotype;
 }
 
-type UnifyError = UnifyOccurCheckError | UnifyMismatchError;
+interface UnifyMissingValueError {
+  type: "unify-missing-value-error";
+  t: Monotype;
```

Comment:
Hm, this way we have lost some information. If we implement it in `unifyArray` as I suggested, we can add both arrays here as `t1` and `t2`. So we capture a bit more of context of where the error happened.

Reply:
Either `t1` or `t2` is `[]`, so there's not much information to gain here. The error message still shows the full expression that failed:

```
Error: file:9:2: Type mismatch

Missing value of type number

(-> number α)

vs.

(-> number number number)


(+ 1)
-^
```

Reply:
Oh, true, because it is recursive. We should probably keep the root objects. Nevermind with the context then.